protocol/testing/e2e/authz/authz_test.go

```diff
@@ -0,0 +1,416 @@
+			subaccounts: []satypes.Subaccount{
+				constants.Alice_Num0_100_000USD,
+				constants.Bob_Num0_100_000USD,
+			},
+
+			msgGrant: &authz.MsgGrant{
+				Granter: constants.AliceAccAddress.String(),
+				Grantee: constants.BobAccAddress.String(),
+				Grant: authz.Grant{
+					Authorization: newAny(
+						authz.NewGenericAuthorization(sdk.MsgTypeURL(&banktypes.MsgUpdateParams{})),
+					),
+				},
+			},
+
+			msgExec: &authz.MsgExec{
+				Grantee: constants.BobAccAddress.String(),
+				Msgs: []*codectypes.Any{
+					newAny(
+						&banktypes.MsgUpdateParams{
+							Authority: lib.GovModuleAddress.String(),
+						},
+					),
+				},
+			},
+
+			expectedMsgExecCheckTxSuccess:   true,
+			expectedMsgExecCheckTxCode:      abcitypes.CodeTypeOK,
+			expectedMsgExecDeliverTxSuccess: false,
+			expectedMsgExecDeliverTxCode:    authz.ErrNoAuthorizationFound.ABCICode(),
+		},
+		"Fail (internal): Bob tries to update gov params (authority = gov)": {
+			subaccounts: []satypes.Subaccount{
+				constants.Alice_Num0_100_000USD,
+				constants.Bob_Num0_100_000USD,
+			},
+
+			msgGrant: nil,
+
+			msgExec: &authz.MsgExec{
+				Grantee: constants.BobAccAddress.String(),
+				Msgs: []*codectypes.Any{
+					newAny(
+						&govtypesv1.MsgUpdateParams{
+							// Authority = gov
+							Authority: lib.GovModuleAddress.String(),
+						},
+					),
+				},
+			},
+
+			expectedMsgExecCheckTxSuccess:   true,
+			expectedMsgExecCheckTxCode:      abcitypes.CodeTypeOK,
+			expectedMsgExecDeliverTxSuccess: false,
+			expectedMsgExecDeliverTxCode:    authz.ErrNoAuthorizationFound.ABCICode(),
+		},
+		"Fail (internal): Bob tries to update gov params (authority = bob)": {
+			subaccounts: []satypes.Subaccount{
+				constants.Alice_Num0_100_000USD,
+				constants.Bob_Num0_100_000USD,
+			},
+
+			msgGrant: nil,
+
+			msgExec: &authz.MsgExec{
+				Grantee: constants.BobAccAddress.String(),
+				Msgs: []*codectypes.Any{
+					newAny(
+						&govtypesv1.MsgUpdateParams{
+							// Authority = bob
+							Authority: constants.BobAccAddress.String(),
+						},
+					),
+				},
+			},
+
+			expectedMsgExecCheckTxSuccess:   true,
+			expectedMsgExecCheckTxCode:      abcitypes.CodeTypeOK,
+			expectedMsgExecDeliverTxSuccess: false,
+			// This fails because Bob is not authorized to create clob pairs (not in the
+			// list of authorities defined in clobkeeper.authorities).
+			expectedMsgExecDeliverTxCode: govtypes.ErrInvalidSigner.ABCICode(),
+		},
+		//
+		// Below tests fail during CheckTx since the ante handler would reject these transactions.
+		//
+		"Fail (app injected): Bob tries to propose operations": {
+			subaccounts: []satypes.Subaccount{
+				constants.Alice_Num0_100_000USD,
+				constants.Bob_Num0_100_000USD,
+			},
+
+			msgGrant: nil,
+
+			msgExec: &authz.MsgExec{
+				Grantee: constants.BobAccAddress.String(),
+				Msgs: []*codectypes.Any{
+					newAny(
+						&clobtypes.MsgProposedOperations{},
```

Comment:
it's both a app injected and also dydx message. we check for app injected before dydx messages so it will get rejected before even reaching dydx message check.